### PR TITLE
Wire in lint and (prepare) error prone to sample

### LINKED ISSUE #293: Integrate Error Prone in sample app
Will be nice to have an example of error prone setup in the sample app.

### PR DESCRIPTION
**Pull request by hzsweers**
A step toward #293 and demos configuration of both.

### CODE REVIEW

**Comment by hzsweers** [comment]
Note that error prone integration right now breaks lint, but my gradle-fu is not strong enough to know what to do about it. My sense is that error-prone is replacing the java compilation task that lint looks for?

```
Caused by: org.gradle.api.artifacts.UnknownConfigurationException: Configuration with name 'compileClasspath' not found.
        at org.gradle.api.internal.artifacts.configurations.DefaultConfigurationContainer.createNotFoundException(DefaultConfigurationContainer.java:137)
        at org.gradle.api.internal.DefaultNamedDomainObjectCollection.getByName(DefaultNamedDomainObjectCollection.java:320)
        at org.gradle.api.internal.artifacts.configurations.DefaultConfigurationContainer.getByName(DefaultConfigurationContainer.java:127)
        at org.gradle.api.internal.artifacts.configurations.DefaultConfigurationContainer.getByName(DefaultConfigurationContainer.java:54)
        at com.android.tools.lint.gradle.LintGradleProject$ProjectSearch.getProject(LintGradleProject.java:984)
        at com.android.tools.lint.gradle.LintGradleProject$ProjectSearch.getProject(LintGradleProject.java:1074)
        at com.android.tools.lint.gradle.LintGradleProject$ProjectSearch.getProject(LintGradleProject.java:876)
        at com.android.tools.lint.gradle.LintGradleClient.createLintRequest(LintGradleClient.java:207)
        at com.android.tools.lint.LintCliClient.run(LintCliClient.java:231)
        at com.android.tools.lint.gradle.LintGradleClient.run(LintGradleClient.java:230)
        at com.android.tools.lint.gradle.LintGradleExecution.runLint(LintGradleExecution.java:291)
        at com.android.tools.lint.gradle.LintGradleExecution.lintAllVariants(LintGradleExecution.java:402)
        at com.android.tools.lint.gradle.LintGradleExecution.analyze(LintGradleExecution.java:96)
        at com.android.tools.lint.gradle.api.ReflectiveLintRunner.runLint(ReflectiveLintRunner.kt:38)
        at com.android.build.gradle.tasks.LintBaseTask.runLint(LintBaseTask.java:88)
        at com.android.build.gradle.tasks.LintGlobalTask.lint(LintGlobalTask.java:47)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)

```

@kageiit @msridhar any idea?

File: sample/gradle.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+autodispose.classes_with_scope=android.app.Activity,android.app.Fragment,com.uber.autodispose.LifecycleScopeProvider,android.arch.lifecycle.LifecycleOwner,androidx.lifecycle.LifecycleOwner,com.uber.autodispose.ScopeProvider,com.uber.autodispose.sample.CustomScope
```
Comment (shaishavgandhi05) [inline sample/gradle.properties]: No need for the android ones since lint is additive and adds android ones by default.

Comment (hzsweers) [reply sample/gradle.properties]: 6d08b33

**Comment by hzsweers** [comment]
CC @tbroyer in case it's potentially an issue with the gradle plugin, but I couldn't find any prior issues like this on the github project

**Pull request merged**